stdpopsim/cli.py

```diff
@@ -432,6 +432,9 @@ def run_simulation(args):
             summarise_usage()
             if ts is not None:
                 write_output(ts, args)
+        elif args.dry_run and args.engine == "slim":
```

Comment:
Potentially a little brittle here? Maybe we should just pass the ``dry_run`` argument through to all the engines and deal with it there.  Then we call ``summarise_usage`` regardless.

Reply:
Yep, that makes loads of sense! In the msprime engine, I've passed `end_time=0` to `msprime.simulate()`. I hope this is a reasonable approach. At the very least, there should be some more thorough parameter value/type checking.